default/scripting/ship_hulls/monster/monster.macros

```diff
@@ -1,3 +1,11 @@
+MONSTER_SHIELD_REGENERATION
+'''EffectsGroup  // Increase shields to max when not in battle
+            scope = Source
+            activation = (Source.LastTurnActiveInBattle < CurrentTurn)
```

Comment:
Should this also test that the ship is unowned to avoid duplicating the effect when an empire owns it?

Reply:
Unowned seems to be the wrong condition.

The critical part is that the ship has no species (as ship shields are regenerated by a species effect). If a player owns it is irrelevant to this. Adding ```Unowned``` condition would actually break a variety of player-owned monsters.

Could check for no species instead, I guess. Or add a stacking group. Or just ignore the possible duplication as it has no noticeable impact.